src/Sql/SqlBase.php

```
@@ -119,6 +119,15 @@ public static function create(array $options = []): ?SqlBase
     public static function getInstance($db_spec, $options): ?self
     {
         $driver = $db_spec['driver'];
+        // Handle custom database drivers.
+        if (!in_array($driver, ['mysql', 'pgsql', 'sqlite'])) {
+            $driver_class = $db_spec['namespace'] . '\\Connection';
+            $connection = (new \ReflectionClass($driver_class))->newInstanceWithoutConstructor();
+            // This will only work if the method is basically static, as most
+            // will be...but we can't truly instantiate the Connection class
+            // here without also calling with a "real" PDO connection.
+            $driver = $connection->databaseType();
```

How about we wrap this in a try/catch and swallow any error.